impeller/tools/malioc.gni

```diff
@@ -0,0 +1,124 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//build/compiled_action.gni")
+import("//flutter/common/config.gni")
+import("//flutter/testing/testing.gni")
+
+declare_args() {
+  # Path to the Mali offline compiler tool 'malioc'.
+  impeller_malioc_path = ""
+
+  impeller_malioc_cores = []
+}
+
+if (impeller_malioc_path != "" && impeller_malioc_cores == []) {
+  core_list_file = "$root_gen_dir/mali_core_list.json"
+  exec_script("//build/gn_run_binary.py",
+              [
+                rebase_path(impeller_malioc_path, root_build_dir),
+                "--list",
+                "--format",
+                "json",
+                "--output",
+                rebase_path(core_list_file),
+              ])
+  _mali_cores = read_file(core_list_file, "json")
+  foreach(mali_core, _mali_cores.cores) {
+    impeller_malioc_cores += [ mali_core.core ]
+  }
+}
+
+template("malioc_analyze_shaders") {
+  # TODO(zra): Check that gles_language_version is in the supported set. For now
```

Comment:
@jonahwilliams ssbos requires a higher language version, so when any language version is set, analysis is skipped. This TODO is for tightening this logic to match what the analyzer supports.

Reply:
makes sense!